pymc/util.py

```diff
@@ -349,3 +350,25 @@ def check_dist_not_registered(dist, model=None):
                 f"You should use an unregistered (unnamed) distribution created via "
                 f"the `.dist()` API instead, such as:\n`dist=pm.Normal.dist(0, 1)`"
             )
+
+
+def point_wrapper(core_function):
```

Comment:
Can we reuse `Point` and `PointFunc`?

https://github.com/pymc-devs/pymc/blob/9a905faa8799f3d4e666603feb7e0e5ba8a8007f/pymc/model.py#L1791-L1822

Reply:
Or improve those, as it seems the main logic is similar

Reply:
I think that the approach I used for the inputs could be copied into `PointFunc` so that it can also handle `*args`, but the main goal of the `point_wrapper` is to only take the values from a `point` which are valid inputs to the core function and ignore the rest

Reply:
Isn't that what `Point(filter_vars=True)`, is trying to do as well, except focused only on `value_vars`?

Reply:
But the point was indeed if we can combine the functionality, instead of keeping two separate things that are kind of trying to do the same

Reply:
The `point_wrapper` is model agnostic, it just cares about the calling function's signature. The `Point` is agnostic of the calling function and only cares about the variables defined in a model. These two things are different, and I'm not sure how would be a simple way to merge them

Reply:
I see Point used only in 1-2 places in the codebase, so it sounds like we can easily transfer the point logic to the `PointFunc` if that makes the two more equivalent.

Reply:
And one of the places used is in SMC for `prior_predictive_sampling` which would probably be addressed by the new changes implemented in the PR